docs/_ext/conversion.py

```diff
@@ -73,57 +73,76 @@ def create_r_returns(doc_list):
 
     return doc_str
 
+pattern_start = re.compile(r'>>> |^\.\.\. ')
+# pattern_methods = re.compile(r'([A-Za-z_]\d*)\.([A-Za-z_]\d*)')
+pattern_methods = re.compile(r'([A-Za-z_]\d*)\.([A-Za-z_]\d*)(?=(?:[^"]|"[^"]*")*$)(?=(?:[^\']|\'[^\']*\')*$)')
+pattern_list_idx = re.compile(r'\w\[(\d+)]')
+pattern_list = re.compile(r'\[([\'\"\w(),\s]+)]')
+pattern_dict = re.compile(r'{(([\'\"]*[\w\d()]+[\'\"]*: [\'\"]*[\w\d]+\'*,*\s*)+)}')
+pattern_doctest = re.compile(r'\s+# doctest:.*')
+
+def line_transpile_py_to_r(line: str) -> str:
+    line = py_to_r_str(line, example=True)
+    line = re.sub(' = ', ' <- ', line)
+
+    # Substitute any . to $, e.g. model.parameters -> model$parameters
+    line = re.sub(pattern_methods, r'\1$\2', line)
+
+    # Substitute """ or ''' to " for multiline strings
+    line = re.sub(r'"""', r'"', line)
+    line = re.sub(r"'''", r'"', line)
+
+    # Check if line has list subscript
+    if re.search(r'\w\[', line):
+        idx = re.search(pattern_list_idx, line)
+        if idx:
+            # Increase idx by 1, e.g. list[0] -> list[1]
+            line = re.sub(idx.group(1), f'{int(idx.group(1)) + 1}', line)
+    else:
+        # Substitute any [] to c(), e.g. ['THETA(1)'] -> c('THETA(1)')
+        line = re.sub(pattern_list, r'c(\1)', line)
+
+    # Check if line contains python dict
+    dict_py = re.search(pattern_dict, line)
+    if dict_py:
+        # Substitute {} to list(), e.g. {'EONLY': 1} -> list('EONLY'=1)
+        dict_r = f'list({dict_py.group(1).replace(": ", "=", )})'
+        line = line.replace('{' + f'{dict_py.group(1)}' + '}', dict_r)
+
+    # Replace len() with length()
+    line = line.replace(r'len(', 'length(')
+
+    # Remove doctest comments
+    line = re.sub(pattern_doctest, '', line)
+    return line
+
+def is_import_line(line: str):
+    return line.startswith('import ') or ' import ' in line
+
+def transpile_py_to_r(lines: List[str]):
+    # must_load_library = any(map(is_import_line, lines))
+    filtered_lines = [line for line in lines if not(is_import_line(line))]
+
+    transpiled_lines = map(
+        lambda line: line_transpile_py_to_r(re.sub(pattern_start, '', line)),
+        filtered_lines,
+    )
+
+    return transpiled_lines
+    # return ['library(pharmr)', *transpiled_lines] if must_load_library else list(transpiled_lines)
```

Comment:
We remove all `import` statements. We assume an implicit `library(pharmr)` has already been run.

Reply:
OK, but most examples in the docs reuse variables from previous examples. Or at least they are run in such a way that it is possible to do so. I see how it makes sense for doctests in the API docs, but not in the general documentation.

Reply:
Agree that if only `pharmpy.modeling` is available in `pharmr` there should be a note about it.

Reply:
Okay, agree! I guess it can be said if you're trying pharmr examples you need to load pharmr, in Python it's more relevant with imports since you can import specific functions. Yes, in the R-documentation there is a note about it